Adapt format of 'Introduction to GitHub' to match 'GitHub Pages & Desktop'

brianamarie opened the pull request:
Currently, the on-demand content for **GitHub Pages and GitHub Desktop** and the content for **Introduction to GitHub** have different formats. 

This PR introduces the topic for conversation - which format is better? Should we adapt the older content to match the newer format? 

I would  ❤️ to have a conversation about this, and form our ideas as commits here to both continue brainstorming and progress. @github/services-training Thoughts on this? 

A few things I'd like to hear from @hollenberry in particular with his ✨ keen eye for data - If we decide to make this transition of format, how could we set ourselves up to gather meaningful data about which format is better received? How could that affect our other content? Is there already something in the 🔢 numbers we should be looking for to inform this decision? 

@github/services-training What do you think about moving the format of the desktop content to the introduction content? Does anyone want to ✋ for this? 

# Final Steps for Shipping

- [x] Finalize how we will introduce the overview video - see [here](https://github.com/github/training-kit/pull/412#discussion_r92412363) and [here](https://github.com/github/training-kit/pull/412#discussion_r92408088)
- [ ] Re-record the overview video **PRP:** @hollenberry
  - ~Add [video bookmarks](https://github.com/github/training-kit/pull/412#issuecomment-266963605) for each step in the process~ Let's discuss this in greater detail since we also have the GIFs - do we want to do both? Could the bookmarks live in the YouTube video description?
- [x] [Create GIFs for each step](https://github.com/github/training-kit/pull/412/commits/0121720153c611c7094120f2186d7e22c80a23da) @brianamarie
- [x] Remove [how to take course from first page](https://github.com/github/training-kit/pull/412#discussion_r92453631) @brianamarie 
- [x] Fix [markdown rendering](https://github.com/github/training-kit/pull/412/commits/00e352ed36a152a29c25a581cb1937bd8be2ab67) @brianamarie

### Other steps (non-blocker)
- Create overlay with removed 'how to take a course' information - Happening in another PR #432

hectorsector commented:
I like the format of the Desktop course -- content is in bite-sized chunks and we've modified the Jekyll templates to standardize the GIF demonstrations and tell-me-whys. The tell-me-whys themselves give the learner the choice to learn more and explore that path on their own without cluttering the page.

hollenberry commented:
> A few things I'd like to hear from @hollenberry in particular with his ✨ keen eye for data - If we decide to make this transition of format, how could we set ourselves up to gather meaningful data about which format is better received? How could that affect our other content? Is there already something in the 🔢 numbers we should be looking for to inform this decision?

The current data dashboard reveals this data:
![screen shot 2016-11-29 at 4 11 10 pm](https://cloud.githubusercontent.com/assets/13326548/20734462/87040712-b64e-11e6-84e6-d48781ca91aa.png)

I think that this is actually an excellent representation of the things that will be most interesting for our future decision making. Users/Sessions/ReturningUsers are all great metrics for how popular our courses are. Additionally, we have a sense of stickiness with Time Spent and Bounce Rate.

I think that there are 3 items missing that we can easily dig up:
- User Flow Percentages (how many students hit the first page compared to those who hit the last page?)
- Pull Requests/Issues within each repository
- Referral Sources 

Using the above data should give us a really good sense of our effectiveness across multiple metrics.

hollenberry commented:
> What do you think about moving the format of the desktop content to the introduction content?

I ❤️ the format of the desktop content. I think we really learned a lot following our first go around that was improved upon. Also, I love @hectorsector 's consistent reminder that modularized content is easier to contribute to in an open source setting. Bite sized pieces can be easily expanded, reviewed, and consistently formatted for syntax and tone throughout the course, which might be difficult in the current state.

brianamarie commented:
> Using the above data should give us a really good sense of our effectiveness across multiple metrics.

@hollenberry ❤️ Thank you for your insights! What sort of trends should we look for if we switch the format? Are there other factors we should keep in mind like new content or trends from our other sites as we look for meaningful takeaways? Do you think there's a certain date we should try to make the shift, or a certain amount of time we should wait from other new content so that the data is clear?

hollenberry commented:
> What sort of trends should we look for if we switch the format?

I think that, generally, we'll just want to see an increase in all of our numbers. Do you have any other ideas for trends? 

> Are there other factors we should keep in mind like new content or trends from our other sites as we look for meaningful takeaways?

We should definitely keep in mind that new content is going to lead to an increase in traffic, but we can still view User Flow for each individual course. Hopefully we have a higher retention rate and we see a change in referral sources, as noted above. 

> Do you think there's a certain date we should try to make the shift, or a certain amount of time we should wait from other new content so that the data is clear?

I think that we should feel clear to make the shift as soon as we're comfortable doing so! As Cynthia mentioned, we can always revert if we're dissatisfied with any unforeseen negative change in trends. We have enough data from these past few months that we'll be able to evaluate our effectiveness on the platform regardless of a new traffic influx. It _should_ all be easy to determine from the User Flow data. 

Thoughts?

brianamarie commented:
> Do you have any other ideas for trends?

@hollenberry Retention rates seem very important. In general, I find it much easier to wrap my head around data when the goal of what we'd like to see is top of mind.

@hollenberry Do you have any interest in attacking these format changes? ⚡️

hollenberry commented:
> Retention rates seem very important. In general, I find it much easier to wrap my head around data when the goal of what we'd like to see is top of mind.

@brianamarie Absolutely. Do you have an idea for these outside of the metrics above? Specifically, Users/Sessions/ReturningUsers. 

> Do you have any interest in attacking these format changes? ⚡️

I'm happy to help out if we decide to tackle this as a team!

brianamarie commented:
> Absolutely. Do you have an idea for these outside of the metrics above? Specifically, Users/Sessions/ReturningUsers.

@hollenberry I don't think I understand the question. 😄 I think we should hope for trends that users go through the entire course instead of dropping off. 

@beardofedu @crichID Thoughts on the format change? I believe @hollenberry & @hectorsector have both given it a theoretical 👍

hollenberry commented:
> I don't think I understand the question. 😄 I think we should hope for trends that users go through the entire course instead of dropping off.

I might be a little lost as well 😄 . You had mentioned retention rates as an "other idea for trends", and I was wondering if I was missing any of the best ways for us to track those. 

> I think we should hope for trends that users go through the entire course instead of dropping off.

Absolutely! 👍

hectorsector commented:
> Do you have any interest in attacking these format changes? ⚡️

As soon as we get 👍 from @crichID and @beardofedu I'm happy to pair up with anyone that wants to take this on and open a PR.

beardofedu commented:
👍 

@hectorsector lets get things done, I kinda want to run through building content in the newer format prior to trying to build a class from scratch, so some hands-on experience might be a good thing 😄 .

crichID commented:
:+1: from me to running this as a controlled experiment. @hollenberry I leave it in your capable hands to set appropriate benchmarks and monitor the stats. Once this launches, would it make sense to put a target date on the calendar for analyzing the stats and making a decision?

hectorsector commented:
> lets get things done

Awesome! Want to start the PR? Or chat synchronously first?

hollenberry commented:
> Once this launches, would it make sense to put a target date on the calendar for analyzing the stats and making a decision?

Yes it would! I'll add a date to my calendar now.

hectorsector commented:
Alright friends! All the content is now using the `simple-class` layout. The next step should be to start combing through the content to ensure that it makes sense as it's presented, and to work through any formatting issues that resulted from the move. 

The general style we've developed is to ensure that each `main-content` section contains just steps with very little superfluous wording. The `tell-me-why` section can be used to provide more in-depth conversations.

crichID commented:
@hectorsector thank you for tackling the lion's share of work in moving this to the new format! I went through and tried to pick up where you left off in getting us down to one focused activity per page. 

@beardofedu @hollenberry would either of you have time to take another pass and look for things like:

- [ ] Formatting consistency between pages
- [ ] Structural consistency
- [ ] Grammar, spelling, punctuation
- [ ] Page naming consistency between Nav and pages
- [ ] Other stuff we need to fix??

It looks like the biggest **To Do** after this will be creating new GIFs for the Show Me sections.

hollenberry commented:
> would either of you have time to take another pass and look for things like:

✋ I'll do this today!

hollenberry commented on paths/intro-to-github/12-merging-pull-requests.md:
```diff
@@ -0,0 +1,47 @@
+
+  When you merge your branch, you are taking the content and history from the branch you created and adding it to the content and history of the `master` branch.
+
+  ![Merge visual]({{site.baseurl}}/images/merge-visual.jpg)
```
This is broken on my end.

brianamarie replied:
@hollenberry I just pushed up a fix. The steps to fix in case you find something similar in the future: 
- Verify the image actually is where it says it is
- Check the path on the link; if it isn't working, look for other working images and emulate that structure. In this case, that went from`{{site.baseurl}}/images/merge-visual.jpg` to `/on-demand/images/merge-visual.jpg`. ✨

hollenberry commented:
Porting over a prior conversation, as I've just added the video with the following adjustments.

- [x] Adding images/animations to the first 30 seconds.
- [x] I'll need to upload to Youtube and ~~then add bookmarks~~. 
- [x] Considering re-recording the first 30 seconds because I sound like the guy from twilight zone.
- [x] Trying to find a way to cut 8 seconds off and be within 3:30

All is complete, except I was unable to add the bookmarks. I'm going to tackle the final step tomorrow with a more concentrated effort. I'm hoping someone knows how to do this. 😄

brianamarie commented:
> Porting over a prior conversation, as I've just added the video with the following adjustments.

@hollenberry The list of video improvements looks great!  [Assuming this link starts at 8:27](https://youtu.be/EwWZbyjDs9c?t=507), the process is pretty simple. [I recommend this Wikihow article for steps](http://www.wikihow.com/Link-to-a-Certain-Time-in-a-YouTube-Video).

brianamarie commented on paths/intro-to-github/00-welcome.md:
```diff
@@ -1,66 +1,54 @@
 
-**GitHub is also a powerful version control tool.**
+  - Use the GitHub.com website to complete the GitHub workflow
+  - Collaborate with ~~strangers~~ new friends on a fun project
```
@hollenberry This isn't striking out for me either. A quick search shows [we may need to alter our configs for Jekyll slightly](http://stackoverflow.com/questions/17004095/jekyll-markdown-strikethrough).

crichID commented on paths/intro-to-github/00-welcome.md:
```diff
@@ -1,66 +1,54 @@
+  <div id="map"></div>
 
-**GitHub is so much more.**
+  ## We Are Here to Help!
```
> What are your thoughts on separating this into a second tab?

@hollenberry by a second tab do you mean a new page? We are already at 15 with the retro, do we want to add more? Could we find a more aesthetically pleasing way to display this information on this page? Maybe we could use something like [this image](http://betterevaluation.org/sites/default/files/StepsforTranslatingEvalFindingstoInfographics.jpg) that describes "steps for successfully completing the course" (e.g. do the activities, connect in gitter, review the tell me why sections, etc).

crichID commented on paths/intro-to-github/14-videorecap.md:
```diff
@@ -0,0 +1,15 @@
+sidebar:
+  nav: "intro-nav"
+main-content: |
+  ## Lightning Video
```
@hollenberry should we add a little more context here? Perhaps something like: "We know it may seem like a lot to remember, so we created a short video of just the steps required to complete the activity."

hectorsector commented on _includes/activity/intro-to-github/04-create-a-file-on-github.md:
```diff
@@ -1,17 +1,14 @@
+        githubHandle: YOUR-USERNAME
+        latitude: YOUR_LATITUDE
+        longitude: YOUR_LONGITUDE
+        ---
```
@hollenberry can't believe you got this to work, thank you!

hollenberry replied:
@hectorsector This is entirely @beardofedu and @brianamarie 's work from Pages. 👍 

I'm pretty thrilled about it, too.

crichID commented:
This one is getting really close! I went through and identified the steps we need to complete before shipping and added them to the [Original Post](https://github.com/github/training-kit/pull/412#issue-191143937). Please put your name next to anything you are PRP'ing and let's try to get this one out before the holidays!

brianamarie commented:
@hollenberry I'd love to help move along the video aspects of this re-format. What can I do to help?

brianamarie commented:
Possible unpopular opinion: 

I am trying to put myself in the shoes of my past-self when I was spending  several hours a day going through online 'learn to code' resources. I'm thinking about where and how to include the recap video so it will be most useful. Right now, it's positioned at the end of the course. I don't think this is a logical place - all of the other steps are completed in order, and that looks like the last step, even though if the users reach that point they wouldn't have any need for the recap. 

We've discussed adding links within the video so we can link to certain portions. However, each individual segment is about ~10 seconds. If we are sending someone help, I imagine it's because the most brief instructions won't cut it and they'll need something more. So in my mind, the ideal audience would be someone who hasn't completed the course but wants a walkthrough of what they're about to see **OR** someone who hasn't seen the course and wants to get more information before committing to doing any work. 

For audience #1, it makes sense to me to just link to an external YouTube page or to include the video in the 'show me how' of the welcome page. 

For audience #2, I think having this video be discoverable and public on YouTube is a good idea. 

I don't mean to prescribe any of this to the course, but rather I would like to better understand the context of this video for our participants. @hollenberry What are your thoughts on this take? Do you think I have an accurate perception of the intended audience for this video?

hectorsector commented:
@brianamarie I'll buttinski here and say that I think the video makes sense near the beginning of the course, and it also makes sense for it to be publicly available. I like to think about our offerings like Tesla thinks about their products... at the end of the day their goal is to *accelerate the world's transition to sustainable energy*. They do so via multiple means: electric cars, solar roofs, commercial-scale batteries for utility companies, self-driving systems, etc. So, if we model ourselves like that, why don't we approach it from every angle at our disposal?

See what I did there, @hollenberry 😉 🚗

brianamarie commented:
> So, if we model ourselves like that, why don't we approach it from every angle at our disposal?

@hectorsector 👍 Sounds good. I've added it to the first page in the 'show me how' section. @hollenberry @hectorsector @github/services-training Thoughts? Is there a better place to put this?

crichID commented:
> I think the video makes sense near the beginning of the course, and it also makes sense for it to be publicly available.

@hectorsector :+1: and :+1:!

I made a couple of tiny commits on the video introduction and it looks like this is pretty much ready to :ship:. Unless someone has an objection, let's plan on pressing the big green button at 4:00 PM EST on Wednesday, December 21st? 

/cc @github/services-training for :eyes: and possible objections??

brianamarie commented:
Small change made - since the video is added to the first page, I removed the last page and updated the link to the video so it now just tells users that it's in the "Show Me How" section.

crichID commented:
@brianamarie thank you for pressing the big green button :tada:!

hollenberry commented:
Hey team! Sorry for the delay: https://youtu.be/kocHKaHtZCc

I've updated the video to cut out "STEP BLANKS" after step 2. I'm happy to swap this in in a fresh new PR, but would love some quick feedback or 👍 before doing so.

hectorsector commented:
Yay! I missed this merge but I'm glad I was able to enjoy it posthumously 💀.

![](http://i.giphy.com/Ch0JvNvkk7PH2.gif)
